riot/riot.py

```diff
@@ -502,25 +647,28 @@ def generate_base_venvs(
     ) -> None:
         """Generate all the required base venvs."""
         # Find all the python interpreters used.
-        required_pys: t.Set[Interpreter] = set(
-            [inst.py for inst in self.venv.instances(pattern=pattern)]
-        )
-        # Apply Python filters.
-        if pythons:
-            required_pys = required_pys.intersection(pythons)
+        required_pys = []
+
+        for inst in self.venv.instances(pattern=pattern):
+            if inst.interpreter is None:
+                continue
```

Comment:
raise?

wondering how we can end up in this state other than misconfiguration?